src/main/app/src/app/informatie-objecten/informatie-objecten.service.ts

```diff
@@ -186,21 +191,25 @@ export class InformatieObjectenService {
     infoObject: EnkelvoudigInformatieObjectVersieGegevens,
   ): Observable<EnkelvoudigInformatieobject> {
     const formData = new FormData();
-    formData.append("uuid", uuid);
-    formData.append("zaakUuid", zaakUuid);
-    formData.append("titel", infoObject.titel);
-    formData.append(
-      "vertrouwelijkheidaanduiding",
-      infoObject.vertrouwelijkheidaanduiding,
-    );
-
-    formData.append("auteur", infoObject.auteur);
-    formData.append("status", infoObject.status);
-    formData.append("taal", JSON.stringify(infoObject.taal));
-    formData.append("bestandsnaam", infoObject.bestandsnaam);
-    formData.append("formaat", infoObject.formaat);
-    formData.append("file", infoObject.file, infoObject.bestandsnaam);
-    formData.append("beschrijving", infoObject.beschrijving);
+    if (uuid) formData.append("uuid", uuid);
```

Comment:
I'm not sure all these `if`s make too much sense because a lot of these fields are mandatory in the (backend) API?

Reply:
Actually it's quite the opposite. None of the fields is mandatory on update and just `titel`, `informatieobjectTypeUUID` and `bestandsnaam` are mandatory on create/upload doc

The bad thing here is that if `field: null` is included in the form the Reader on backend side will break as it cannot really de-serialize string `null`

Reply:
What about using a function that checks an object field and adds the field to the form instead? Something like:
```
const appendToForm = (formData, obj, key: String) => {
    if obj[key] formData.append(key, obj[key])
}
```

Reply:
Ok, clear. Yeah, I would definitely not add null checking to the backend. Sending nulls in API requests is just bad practise which we should avoid.

Reply:
I've looked into this for a bit and this code is not just checking for nulls, there is also some minor mapping going on (for example 'bestand' => 'file'). I'd much rather solve this with the generated client but it doesn't handle formdata very well. Let's discuss tomorrow because I think there are better options for generated clients. For now, I will see if I can simplify this code a bit.

Reply:
@edgarvonk @bas-info-nl @hsiliev Looking to the future I would like to move to another generated client. We only use the current generated client in `identity.service.ts` so the refactor wouldn't be too big, but I don't think it makes sense to do it in the scope of this story. If you are happy with the change I made then I think this one is ready to merge :-)